handsanitizer/FunctionCallerGenerator.cpp

```diff
@@ -0,0 +1,223 @@
+//
+// Created by sabastiaan on 22-07-21.
+//
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include <iomanip>
+#include "FunctionCallerGenerator.h"
+
+namespace handsanitizer {
+    void FunctionCallerGenerator::generate_cpp_file_for_function(std::string dest_file_path) {
```

Comment:
Use `const` and references wherever you don't need mutability or your own copy of the object:

```suggestion
    void FunctionCallerGenerator::generate_cpp_file_for_function(const std::string& dest_file_path) {
```